Java/libraries/recognizers-text-date-time/src/main/java/com/microsoft/recognizers/text/datetime/extractors/BaseTimeExtractor.java

```diff
@@ -19,14 +28,120 @@ public String getExtractorName() {
     }
+
+    private List<ExtractResult> MergeTimeZones(List<ExtractResult> timeErs, List<ExtractResult> timeZoneErs, String text)
+    {
+        for (ExtractResult er : timeErs)
+        {
+            for (ExtractResult timeZoneEr :timeZoneErs)
+            {
+                int begin = er.start + er.length;
+                int end = timeZoneEr.start;
+
+                if (begin > end)
+                {
+                    String gapText = text.substring(begin, end - begin);
+
+                    if (StringUtility.isNullOrWhiteSpace(gapText))
+                    {
+                        int newLenght = timeZoneEr.start + timeZoneEr.length - er.start;
+
+                        er = er.withText(text.substring(er.start, newLenght));
+                        er = er.withLength(newLenght);
+                        er = er.withData(new AbstractMap.SimpleEntry<>(Constants.SYS_DATETIME_TIMEZONE, timeZoneEr));
+                    }
+                }
+            }
+        }
+        return timeErs;
+    }
+
+    public final List<Token> BasicRegexMatch(String text)
+    {
+        List<Token> ret = new ArrayList<>();
+        for (Pattern regex : this.config.getTimeRegexList())
+        {
+            Match[] matches = RegExpUtility.getMatches(regex, text);
+            for (Match match : matches)
+            {
+                ret.add(new Token(match.index, match.index + match.length));
+            }
+        }
+        return ret;
+    }
+
+    private List<Token> AtRegexMatch(String text)
+    {
+        List<Token> ret = new ArrayList<>();
+        // handle "at 5", "at seven"
+        Pattern pattern = this.config.getAtRegex();
+        Match[] matches = RegExpUtility.getMatches(pattern, text);
+        if (matches.length > 0)
+        {
+            for (Match match : matches)
+            {
+                if (match.index + match.length < text.length() && text.charAt(match.index + match.length) == '%')
+                {
+                    continue;
+                }
+                ret.add(new Token(match.index, match.index + match.length));
+            }
+        }
+        return ret;
+    }
+
+    private List<Token> BeforeAfterRegexMatch(String text)
+    {
+        List<Token> ret = new ArrayList<>();
+        // only enabled in CalendarMode
+        if (this.config.getOptions() ==  DateTimeOptions.CalendarMode)
+        {
+            // handle "before 3", "after three"
+            Pattern beforeAfterRegex = this.config.getTimeBeforeAfterRegex();
+            Match[] matches = RegExpUtility.getMatches(beforeAfterRegex, text);
+            if (matches.length > 1)
+            {
+                for (Match match : matches)
+                {
+                    ret.add(new Token(match.index, match.index + match.length));
+                }
+            }
+        }
+        return ret;
+    }
+
+    private List<Token> SpecialCasesRegexMatch (String text)
+    {
```

Comment:
NIT: Code standard. Curly brace should be on the same line as the declaration. 👮 (And all the above functions and methods)